pkg/network/master/subnets.go

```diff
@@ -190,53 +190,44 @@ func (master *OsdnMaster) deleteNode(nodeName string) error {
 // TODO: make upstream kubelet more flexible with overlays and GCE so this
 // condition doesn't get added for network plugins that don't want it, and then
 // we can remove this function.
-func (master *OsdnMaster) clearInitialNodeNetworkUnavailableCondition(node *kapi.Node) {
+func (master *OsdnMaster) clearInitialNodeNetworkUnavailableCondition(origNode *kapi.Node) {
+	// Informer cache should not be mutated, so get a copy of the object
+	node := origNode.DeepCopy()
 	knode := node
 	cleared := false
 	resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
 		var err error
 
 		if knode != node {
-			knode, err = master.kClient.Core().Nodes().Get(node.ObjectMeta.Name, metav1.GetOptions{})
+			knode, err = master.kClient.Core().Nodes().Get(node.Name, metav1.GetOptions{})
 			if err != nil {
 				return err
 			}
 		}
 
-		// Let caller modify knode's status, then push to api server.
-		_, condition := GetNodeCondition(&node.Status, kapi.NodeNetworkUnavailable)
-		if condition != nil && condition.Status != kapi.ConditionFalse && condition.Reason == "NoRouteCreated" {
-			condition.Status = kapi.ConditionFalse
-			condition.Reason = "RouteCreated"
-			condition.Message = "openshift-sdn cleared kubelet-set NoRouteCreated"
-			condition.LastTransitionTime = metav1.Now()
-			knode, err = master.kClient.Core().Nodes().UpdateStatus(knode)
-			if err == nil {
-				cleared = true
+		for i := range knode.Status.Conditions {
+			if knode.Status.Conditions[i].Type == kapi.NodeNetworkUnavailable {
+				condition := &knode.Status.Conditions[i]
```

Comment:
for _, condition := range knode.Status.Conditions {
            if condition.Type == kapi.NodeNetworkUnavailable {